src/psyclone/psyir/nodes/call.py

```diff
@@ -750,70 +766,63 @@ def _get_argument_routine_match(self, routine: Routine):
             # TODO #759: Optional keyword is not yet supported in psyir.
             # Hence, we use a simple string match.
             if ", OPTIONAL" not in str(routine_arg.datatype):
+                routine_arg_name = routine_arg.name.replace("\n", "")
+                routine_name = routine.name.replace("\n", "")
+                call_name = self.debug_string().replace("\n", "")
                 raise CallMatchingArgumentsNotFound(
-                    f"Argument '{routine_arg.name}' in subroutine"
-                    f" '{routine.name}' does not match any in the call"
-                    f" '{self.debug_string()}' and is not OPTIONAL."
+                    f"Argument '{routine_arg_name}' in subroutine"
+                    f" '{routine_name}' does not match any in the call"
+                    f" '{call_name}' and is not OPTIONAL."
                 )
 
         return ret_arg_idx_list
 
     def get_callee(
-        self,
-        check_matching_arguments: bool = True,
-    ):
+            self,
+            use_first_callee_and_no_arg_check: bool = False
+    ) -> Tuple[Routine, List[int]]:
         '''
         Searches for the implementation(s) of the target routine for this Call
         including argument checks.
 
-        If `check_matching_arguments` is set to `False`, the very first
-        implementation of the matching routine will be returned in case no
-        match was found. Then, the arguments of the call and routine
-        might not match each other.
-
-        :param check_matching_arguments: Also check argument types to match.
-            If set to `False` and in case it doesn't find matching arguments,
-            the very first implementation of the matching routine will be
-            returned (even if the argument type check failed). The argument
-            types and number of arguments might therefore mismatch!
-        :type ret_arg_match_list: bool
+        :param use_first_callee_and_no_arg_check: whether or not (the default)
```

Comment:
Could you leave the original content of the doc string, I think it might be useful to highlight the potential risks:

> if `use_first_callee_and_no_arg_check` is set to ... might not match each other

Reply:
Thanks